lcls-twincat-common-components/Library/Devices/PPM/FB_PPM_PowerMeter.TcPOU

```diff
@@ -28,61 +40,110 @@ VAR
     fVoltageBuffer: ARRAY[1..1000] OF LREAL;
 
     {attribute 'pytmc' := '
-        pv: CALIB
+        pv: STC
         io: input
     '}
-    fCalibBase: LREAL;
+    fbTempSensor: FB_CC_TempSensor;
 
     {attribute 'pytmc' := '
-        pv: CALIB_BUFFER
-        io: input
+        pv: AccTypeFromEpics
+        link: TPR:FEES:GD02:0:MODE.RVAL
     '}
-    fCalibBaseBuffer: ARRAY[1..1000] OF LREAL;
+    uAccelerator: UINT;
 
     {attribute 'pytmc' := '
-        pv: MJ
-        io: input
+        pv: NCBeamRateFromEpics
+        link: TPG:SYS0:1:DST04:RATE
+    '}
+    uNCFrequency: UINT;
+
+    {attribute 'pytmc' := '
+        pv: SCBeamRateFromEpics
+        link: TPG:SYS0:1:DST04:RATE
+    '}
+    uSCFrequency: UINT;
+
+    uFrequency: UINT;
+
+    // Pulse-by-pulse energy
+    {attribute 'pytmc' := '
+        pv: PulseEnergy
+        io: i
         field: EGU mJ
     '}
     fCalibMJ: LREAL;
 
+    // Pulse-by-pulse energy buffer
     {attribute 'pytmc' := '
-        pv: MJ_BUFFER
-        io: input
+        pv: PulseEnergy_BUFFER
+        io: i
         field: EGU mJ
     '}
     fCalibMJBuffer: ARRAY[1..1000] OF LREAL;
 
+    // Wattage
     {attribute 'pytmc' := '
-        pv: STC
-        io: input
+        pv: PulseWattage
+        io: i
+        field: EGU W
     '}
-    fbTempSensor: FB_CC_TempSensor;
+    fPulseWattage: LREAL;
 
+    // Background voltage
     {attribute 'pytmc' := '
-        pv: CALIB:OFFSET
+        pv: BackgroundVoltage
         io: io
+        field: EGU V
     '}
-    fCalibRelOffset: LREAL;
+    fBackgroundVoltage: LREAL;
+
+    fBackgroundVoltageAcc: LREAL;
+    fBackgroundVoltageSum: LREAL;
+    fBackgroundVoltageStale: LREAL;
+    tonBackgroundAutoCollecting: TON;
+    fBackgroundVoltageBuffer: ARRAY[1..100] OF LREAL;
+    udBackgroundVoltageBufferIndex: UDINT := 1;
+    i: UDINT;
 
     {attribute 'pytmc' := '
-        pv: CALIB:RATIO
+        pv: BackgroundMode
         io: io
     '}
-    fCalibRelRatio: LREAL;
+    BACKGROUND_MODE: (Manual, Passive) := Passive;
```

Comment:
Is this some sort of inline enum definition? I hadn't seen this before

Reply:
I think I found it here and decided to use it since it was pretty specific. https://alltwincat.com/2021/11/16/local-enumerations/
I could make a normal DUT type if that'd be clearer.

Reply:
Nah, keep it in, this is great and I'm happy you've shown me how to use it.